src/assets/scripts/aircraft/AircraftInstanceModel.js

```diff
@@ -1995,101 +1994,115 @@ export default class Aircraft {
+            glideslope_altitude = _clamp(0, runway.getGlideslopeAltitude(offset[1]), this.altitude);
+            const assignedHdg = this.fms.currentWaypoint().heading;
+            const localizerRange = runway.ils.enabled ? runway.ils.loc_maxDist : 40;
             this.offset_angle = offset_angle;
             this.approachOffset = abs(offset[0]);
             this.approachDistance = offset[1];
-            angle = runway.angle;
-
-            if (angle > tau()) {
-                angle -= tau();
-            }
-
-            glideslope_altitude = _clamp(0, runway.getGlideslopeAltitude(offset[1]), this.altitude);
-            glideslope_window   = abs(runway.getGlideslopeAltitude(offset[1], degreesToRadians(1)));
+            this.target.heading = assignedHdg;
+            this.target.turn = this.fms.currentWaypoint().turn;
+            this.target.altitude = this.fms.currentWaypoint().altitude;
+            this.target.speed = this.fms.currentWaypoint().speed;
 
+            // Established on ILS
             if (this.mode === FLIGHT_MODES.LANDING) {
-                this.target.altitude = glideslope_altitude;
-            }
+                // Final Approach Heading Control
+                const severity_of_correction = 25;  // controls steepness of heading adjustments during localizer tracking
+                const tgtHdg = angle + (offset_angle * -severity_of_correction);
+                const minHdg = angle - degreesToRadians(30);
+                const maxHdg = angle + degreesToRadians(30);
+                this.target.heading = _clamp(tgtHdg, minHdg, maxHdg);
 
-            let ils = runway.ils.loc_maxDist;
-            if (!runway.ils.enabled || !ils) {
-                ils = 40;
-            }
+                // Final Approach Altitude Control
+                this.target.altitude = Math.min(this.fms.currentWaypoint().altitude, glideslope_altitude);
 
-            // lock ILS if at the right angle and altitude
-            if ((abs(this.altitude - glideslope_altitude) < glideslope_window)
-                && (abs(offset_angle) < degreesToRadians(10))
-                && (offset[1] < ils)
-            ) {
-                if (abs(offset[0]) < 0.05 && this.mode !== FLIGHT_MODES.LANDING) {
-                    this.mode = FLIGHT_MODES.LANDING;
+                // Final Approach Speed Control
+                if (this.fms.currentWaypoint().speed > 0) {
+                    this.fms.setCurrent({ start_speed: this.fms.currentWaypoint().speed });
+                }
+                if (this.isLanded()) {
+                    this.target.speed = 0;
+                } else {
+                    const dist_final_app_spd = 3.5; // 3.5km ~= 2nm
+                    const dist_assigned_spd = 9.5;  // 9.5km ~= 5nm
+                    this.target.speed = extrapolate_range_clamp(dist_final_app_spd, offset[1], dist_assigned_spd,
+                                            this.model.speed.landing, this.fms.currentWaypoint().start_speed);
```

Comment:
when parameters make a function/method call too long, separate the parameters on a single line:

```javascript
this.target.speed = extrapolate_range_clamp(
    dist_final_app_spd, offset[1],
    dist_assigned_spd,
    this.model.speed.landing,
    this.fms.currentWaypoint().start_speed
);
```